runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -127,7 +185,14 @@ class ExecutionContext final {
 
     [[nodiscard]] auto getDeviceRecorderStatus() const -> bool { return _tape_recording; }
 
-    [[nodiscard]] bool initDevice(std::string_view name) noexcept
+    [[nodiscard]] QuantumDevice *loadDevice(std::string filename)
+    {
+        _driver_so_ptr = std::make_unique<SharedLibraryManager>(filename);
+        void *f_ptr = _driver_so_ptr->getSymbol("getCustomDevice");
+        return f_ptr ? reinterpret_cast<decltype(getCustomDevice) *>(f_ptr)() : nullptr;
```

Comment:
Does the pointer not get deleted at the end of the scope, and therefore the library closed?

Reply:
`f_ptr` is alive from `dlsym` all the way to `dlclose` because it is just a function pointer. It is not an object. The library is not closed at the end of this scope. The library closes when the destructor of the SharedLibraryManager is called which is not at the end of this function. I don't think it makes sense to unload the library at this particular program point because we still need to be able to execute the whatever implementation of the QuantumDevice, which will depend on functions on the shared library.

Reply:
Just to be more explicit, the variable `f_ptr` is not alive after the end of this scope, but the memory pointed to by `f_ptr` is still valid.